crypto/http/http_client.c

```diff
@@ -964,11 +981,20 @@ OSSL_HTTP_REQ_CTX *OSSL_HTTP_open(const char *server, const char *port,
                             buf_size, overall_timeout);
 
  end:
-    if (rctx != NULL)
+    if (rctx != NULL) {
         /* remove any spurious error queue entries by ssl_add_cert_chain() */
         (void)ERR_pop_to_mark();
-    else
+#if !defined(OPENSSL_NO_TRACE) && !defined(OPENSSL_NO_STDIO)
+        {
+            BIO *trc_out = BIO_new_fp(stdout, BIO_NOCLOSE);
+
+            if (!OSSL_trace_set_channel(OSSL_TRACE_CATEGORY_HTTP, trc_out))
+                BIO_free(trc_out);
+        }
+#endif
```

Comment:
I do not think this is the right thing to do here. IMO that should be done by an application only. The openssl application can already enable tracing output.

Reply:
Yep, removed this.